### PR TITLE
Added ESLint Flat Config

### PR DESCRIPTION
**Pull request by The-CodeINN**
# ESLint 9 + ESLint Flat Config Migration

## Overview
This PR migrates our ESLint configuration to the new ESLint 9 flat config format and updates our linting setup for better performance and maintainability.

## Key Changes
- Migrated from legacy ESLint config to new flat config system
- Updated ESLint configuration to use ES modules instead of CommonJS
- Improved ignore patterns to prevent unnecessary file scanning
- Added caching support for faster lint runs
- Consolidated lint scripts for better consistency

### Configuration Updates
- Replaced `.eslintrc.cjs,` with `eslint.config.mjs` using flat config format
- Implemented proper TypeScript and Astro file handling
- Added explicit file type matching for better control

### Script Updates
```json
"scripts": {
  "lint": "eslint --cache \"src/**/*.{js,ts,astro}\" \"*.{js,ts}\"",
  "lint:fix": "eslint --cache --fix \"src/**/*.{js,ts,astro}\" \"*.{js,ts}\""
}
```

## Benefits
- **Improved Performance**: Added caching support for faster subsequent lint runs
- **Better Type Safety**: Enhanced TypeScript integration with explicit parser configuration
- **Cleaner Configuration**: Modern ES module syntax and flat config structure
- **Reduced Complexity**: Consolidated ignore patterns and file matching

## Testing Done
- Verified all existing files are properly linted
- Confirmed TypeScript and Astro files are correctly parsed
- Tested both `lint` and `lint:fix` commands
- Validated ignore patterns are working as expected

## How to Test
1. Pull the branch
2. Run `npm install` to update dependencies
3. Try `npm run lint` to check files
4. Try `npm run lint:fix` to test auto-fixing
5. Verify no unwanted files (like `node_modules`) are being linted

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "**/node_modules/**",
+        "**/dist/**",
+        "**/package-lock.json",
+        "**/*.md",
+        "**/*.min.js",
+        "content/**/*",
+        "public/content/**/*",
+        ".git/**",
+        ".vscode/**",
+        "coverage/**"
```
Comment (crutchcorn) [inline eslint.config.mjs:28]: Why are we covering these three lines? `.git` I could see an argument for, but `.vscode` and `coverage` likely shouldn't be there.

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Hi! Thanks for the PR!

This PR is missing a few things, namely:
- Deleting `.eslintrc.cjs`
- Deleting `.eslintignore`
- Removing `FlatCompat`

Once these are done, we can merge this :)

**Comment by The-CodeINN** [comment]
I ran  `npx @eslint/migrate-config.eslintrc.cjs` based on their guide for migration https://eslint.org/docs/latest/use/configure/migration-guide and made some changes.

Nonetheless, I'd make the changes as you had recommended and get back soon.

**Comment by The-CodeINN** [comment]
@crutchcorn I have made the recommended changes.

- Remove ESLint compatibility layer (FlatCompat)
- Use typescript-eslint's new flat config setup with tseslint.config()
- Maintain Astro and existing rule configurations
- Update parser configurations for better type checking

**Comment by crutchcorn** [comment]
What a glowup change! Thank you! 😁🚀 I'll be continuing to fix linting errors before merging, but as far as I'm concerned; this is a successful PR!

BTW, I'd love to invite you to list yourself as a developer on our site and join our Discord if you'd like to join us more long-term:

https://discord.gg/FMcvc6T

https://github.com/playfulprogramming/playfulprogramming/blob/main/CONTRIBUTING.md#creating-an-author-profile (just change `author` to `developer`)

**Comment by The-CodeINN** [comment]
> What a glowup change! Thank you! 😁🚀 I'll be continuing to fix linting errors before merging, but as far as I'm concerned; this is a successful PR!
> 
> BTW, I'd love to invite you to list yourself as a developer on our site and join our Discord if you'd like to join us more long-term:
> 
> https://discord.gg/FMcvc6T
> 
> https://github.com/playfulprogramming/playfulprogramming/blob/main/CONTRIBUTING.md#creating-an-author-profile (just change `author` to `developer`)

Thank you 🤩

**Pull request merged**